cmd/commands/runtime.go

```diff
@@ -204,24 +203,38 @@ func NewRuntimeInstallCommand() *cobra.Command {
 	return cmd
 }
 
-func getComponents(rt *runtime.Runtime, opts *RuntimeInstallOptions) []string {
-	var componentNames []string
+func getComponents(rt *runtime.Runtime, opts *RuntimeInstallOptions) []*string {
+	var componentNames []*string
 	for _, component := range rt.Spec.Components {
-		componentNames = append(componentNames, fmt.Sprintf("%s-%s", opts.RuntimeName, component.Name))
+		componentFullName := fmt.Sprintf("%s-%s", opts.RuntimeName, component.Name)
+		componentNames = append(componentNames, &componentFullName)
 	}
 
 	//  should find a more dynamic way to get these additional components
 	additionalComponents := []string{"events-reporter", "workflow-reporter"}
 	for _, additionalComponentName := range additionalComponents {
-		componentNames = append(componentNames, fmt.Sprintf("%s-%s", opts.RuntimeName, additionalComponentName))
+		componentFullName := fmt.Sprintf("%s-%s", opts.RuntimeName, additionalComponentName)
+		componentNames = append(componentNames, &componentFullName)
 	}
-	componentNames = append(componentNames, "argo-cd")
+	argoCDFullName := "argo-cd"
+	componentNames = append(componentNames, &argoCDFullName)
 
 	return componentNames
 }
 
-func createRuntimeOnPlatform(ctx context.Context, runtimeName string, server string, runtimeVersion string, ingressHost string, componentNames []string) (string, error) {
-	runtimeCreationResponse, err := cfConfig.NewClient().V2().Runtime().Create(ctx, runtimeName, server, runtimeVersion, ingressHost, componentNames)
+func createRuntimeOnPlatform(ctx context.Context, opts *model.RuntimeInstallationArgs) (string, error) {
+
+	runtimeCreationResponse, err := cfConfig.NewClient().V2().Runtime().Create(ctx, &model.RuntimeInstallationArgs{
+		RuntimeName:    opts.RuntimeName,
+		Cluster:        opts.Cluster,
+		RuntimeVersion: opts.RuntimeVersion,
+		ComponentNames: opts.ComponentNames,
+		IngressHost:    opts.IngressHost,
+	})
+
+	if err != nil {
```

Comment:
this is not done correctly
What if `ErrorMessage` is not empty but `err` is nil? and vice versa

Reply:
the intention is that:
* if err is nil - it means the graphql request returned with 200. so this is in order to catch bad requests.
* if ErrorMessage is not empty - it means that while the request was good, there was an error while trying to create one of the placeholder entities (runtime, appProject, components), and the ErrorMessage will be descriptive according to which step failed.